### DISCUSSION TITLE
Cannot start the service using GCP RUN or Docker

### DISCUSSION THREAD

**Discussion by annisat** [fastapi/fastapi#11865] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()

@app.get("/")
async def root():
    return {"status": 0, "msg": "API is running"}
```

### Description

By title this looks like https://github.com/tiangolo/fastapi/issues/5946 but is not.
I was migrating my APIs to new CLI `pipenv run fastapi run app\main.py` and it works fine in my local Windows environment. But the Dockerfile doesn't work.
```
FROM python:3.10.13-slim
WORKDIR /cwd
COPY requirements_reduced.txt ./
RUN pip install --no-cache-dir -r requirements_reduced.txt
COPY app/reduced.py ./app/
EXPOSE 8000
CMD ["fastapi", "run", "app/reduced.py"]
```

It says
```
docker: Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "fastapi": executable file not found in $PATH: unknown.
```

I did a little digging by ls everything in $PATH. `ls -al /usr/local/bin` yields:
```
drwxr-xr-x 1 root root  4096 Jul 19 10:48 .
drwxr-xr-x 1 root root  4096 Mar 11 00:00 ..
lrwxrwxrwx 1 root root     9 Mar 12 12:24 2to3 -> 2to3-3.10
-rwxr-xr-x 1 root root   102 Mar 12 12:24 2to3-3.10
lrwxrwxrwx 1 root root     5 Mar 12 12:24 idle -> idle3
lrwxrwxrwx 1 root root     8 Mar 12 12:24 idle3 -> idle3.10
-rwxr-xr-x 1 root root   100 Mar 12 12:24 idle3.10
-rwxr-xr-x 1 root root   238 Jul 19 10:48 normalizer
-rwxr-xr-x 1 root root   226 Mar 12 12:24 pip
-rwxr-xr-x 1 root root   226 Mar 12 12:24 pip3
-rwxr-xr-x 1 root root   226 Mar 12 12:24 pip3.10
lrwxrwxrwx 1 root root     6 Mar 12 12:24 pydoc -> pydoc3
lrwxrwxrwx 1 root root     9 Mar 12 12:24 pydoc3 -> pydoc3.10
-rwxr-xr-x 1 root root    85 Mar 12 12:24 pydoc3.10
lrwxrwxrwx 1 root root     7 Mar 12 12:24 python -> python3
lrwxrwxrwx 1 root root    14 Mar 12 12:24 python-config -> python3-config
lrwxrwxrwx 1 root root    10 Mar 12 12:24 python3 -> python3.10
lrwxrwxrwx 1 root root    17 Mar 12 12:24 python3-config -> python3.10-config
-rwxr-xr-x 1 root root 14472 Mar 12 12:24 python3.10
-rwxr-xr-x 1 root root  3048 Mar 12 12:24 python3.10-config
-rwxr-xr-x 1 root root   216 Jul 19 10:48 uvicorn
-rwxr-xr-x 1 root root   213 Mar 12 12:24 wheel
```
There is `uvicorn` but no `fastapi`. Any idea where pip could have installed the fastapi executable?

### Operating System

Linux, Windows

### Operating System Details

Works fine with my local Windows.
Fails in GCP Run and my local WSL Linux environment.
The image I use is `python:3.10.13-slim`

### FastAPI Version

fastapi==0.110.0

### Pydantic Version

pydantic==2.6.1, pydantic-core==2.16.2

### Python Version

python:3.10.13

**Comment by YuriiMotov** [comment, marked as answer]
FastAPI-cli was added in 0.111.0.
Since you are using `fastapi==0.110.0` you can't use `CMD ["fastapi", "run", "app/reduced.py"]`.
Upgrade to the latest version of FastAPI

**Reply by annisat** [reply]
Thanks! That works!

What still puzzles me, though, is my local Pipfile.lock. It's
```
"fastapi": {
            "hashes": [
                "sha256:266775f0dcc95af9d3ef39bad55cff525329a931d5fd51930aadd4f428bf7ff3",
                "sha256:87a1f6fb632a218222c5984be540055346a8f5d8a68e8f6fb647b1dc9934de4b"
            ],
            "index": "pypi",
            "version": "==0.110.0"
        },
```
But `pipenv run fastapi run app\reduced.py` works just fine.
